manifest-build-tools/application/manifest_generator.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, EMC, Inc.
+
+"""
+The script generate a new manifest for a new branch according to another manifest
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/manifest_generator.py \
+--source-manifest build-manifest/rackhd-devel \
+--dest-manifest build-manifest/rackhd-release-1.2.5 \
+--branch branch/release-1.2.5 \
+--force \
+--publish \
+--publish-branch master \
+--git-credential https://github.com,GITHUB
+
+The required parameters: 
+source-manifest: The path of manifest. The new manifest is generated according to it.
+dest-manifest: The path of the new manifest
+branch: The new branch name
+
+The optional parameters:
+force: If true, overwrite the destination manifest file even it already exists.
+publish: If true, the script will try to push the new manifest to github. That means the dest manifest should         under the manifest repository.
```

Comment:
extra spaces.. and what does it mean "dest manifest should under the manifest repo "?
if the dest-manifest is a new file name, will you create a new file ?